lib/core/dispatchRequest.js

```diff
@@ -52,12 +53,23 @@ module.exports = function dispatchRequest(config) {
   return adapter(config).then(function onAdapterResolution(response) {
     throwIfCancellationRequested(config);
 
-    // Transform response data
-    response.data = transformData(
-      response.data,
-      response.headers,
-      config.transformResponse
-    );
+    try {
+      // Transform response data
+      response.data = transformData(
+        response.data,
+        response.headers,
+        config.transformResponse
+      );
+    } catch (e) {
```

Comment:
Maybe we can throw the error directly.

Reply:
Regarding #2856 
I think in this case, we will create some breaking changes for users regardless which PRs we choose. Both look like fine solutions to me. IMO, this PR might have a smaller impact? (i could be wrong)

> Maybe we can throw the error directly.

If we don't use `createError`, the user will have to figure out all the context of this error by themselves. I was following the tradition in `adapters/http.js` and `adapters/xhr`, but yeah please let me know if you still want me to change it.